### PR TITLE
Actually phases out noteleport

### PR DESCRIPTION
**Pull request by SomeguyManperson**
# Document the changes in your pull request

replaces all mentions of var/noteleport with the area flag since that was half done then forgotten about for probably three months

# Why is this good for the game?
Behavior for teleport restriction was sporadic since either noteleport or area_flags could get checked and sometimes not both

# Testing
Please do, I don't want to have to downgrade my byond again

# Changelog

:cl: Code Janitor Theos
bugfix: teleport restricting areas now restrict teleporting reliably
/:cl:

### CODE REVIEW

File: code/game/area/areas/ruins/lavaland.dm [hunk 1]
```diff
@@ -69,12 +69,12 @@
 
 /area/ruin/powered/kinggoat_arena //yogs start
 	name = "King Goat Arena"
-	noteleport = TRUE
+	area_flags = NOTELEPORT
```
Comment (JohnFulpWillard) [inline code/game/area/areas/ruins/lavaland.dm:72]: Does parent have any flags that they should keep? (This applies to all)
```suggestion
	area_flags = parent_type::area_flags | NOTELEPORT
```

Comment (Moltijoe) [reply code/game/area/areas/ruins/lavaland.dm:72]: wait, you can do that?

Comment (SomeguyManperson) [reply code/game/area/areas/ruins/lavaland.dm:72]: None of the other area flags are used outside lavaland generation so no but I can do this anyways

**Review by Aquizit — APPROVED** [review]
map thing seems fine

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**